make_string(LIBCURL_VERSION) fails clang's Wstring-conversion

https://github.com/jeroen/curl/blob/ae452a8dd326fd49daf8e1352fbd4da8a8f05a77/src/version.c#L10

Gives compilation failure on clang (https://clang.llvm.org/docs/DiagnosticsReference.html#wstring-conversion):

```
src/version.c:10:39: error: implicit conversion turns string literal into bool: 'char[11]' to '_Bool' [-Werror,-Wstring-conversion]
   10 |   SET_VECTOR_ELT(list, 1, make_string(LIBCURL_VERSION));
      |                           ~~~~~~~~~~~~^~~~~~~~~~~~~~~~
$PATH/curl/src/include/curl/curlver.h:35:25: note: expanded from macro 'LIBCURL_VERSION'
   35 | #define LIBCURL_VERSION "8.10.1-DEV"
      |                         ^~~~~~~~~~~~
src/curl-common.h:11:24: note: expanded from macro 'make_string'
   11 | #define make_string(x) x ? Rf_mkString(x) : Rf_ScalarString(NA_STRING)
      |                        ^ ~
```

This looks reasonable to me -- char literal on LHS of `?` looks questionable. I haven't read all the other uses of `make_string()` carefully enough, but maybe something like

```c
#define make_string(x) x!=NULL && x ? Rf_mkString(x) : Rf_ScalarString(NA_STRING)
```

Would be better? That compiles for me. Maybe just `x != NULL ? ...` is enough.

--- a/src/curl-common.h
+++ b/src/curl-common.h
@@ -8,7 +8,7 @@
 #include <string.h>
 #include <stdlib.h>
 
-#define make_string(x) x ? Rf_mkString(x) : Rf_ScalarString(NA_STRING)
+#define make_string(x) x != NULL ? Rf_mkString(x) : Rf_ScalarString(NA_STRING)
 #define get_string(x) CHAR(STRING_ELT(x, 0))
 #define assert(x) assert_message(x, NULL)
 

